### PR TITLE
Rid 5053 transmit security connector

### PR DESCRIPTION
**Pull request by tomzarhin5**
Required items, please complete
   
   Change(s):
   - See guidance below

   Reason for Change(s):
   - See guidance below

   Version Updated:
   - Required only for Detections/Analytic Rule templates
   - See guidance below

   Testing Completed:
   - See guidance below

   Checked that the validations are passing and have addressed any issues that are present:
   - See guidance below

# Guidance <- remove section before submitting
-----------------------------------------------------------------------------------------------------------
## Before submitting this PR please ensure that you have read the following sections and filled out the changes, reason for change and testing complete sections:

Thank you for your contribution to the Microsoft Sentinel Github repo.

> Details of the code changes in your submitted PR.  Providing descriptions for pull requests ensures there is context to changes being made and greatly enhances the code review process.  Providing associated Issues that this resolves also easily connects the reason.
   
   Change(s):
   - Updated syntax for XYZ.yaml

   Reason for Change(s):
   - New schema used for XYZ.yaml
   - Resolves ISSUE #1234

   Version updated:
   - Yes
   - Detections/Analytic Rule templates are required to have the version updated

> The code should have been tested in a Microsoft Sentinel environment that does not have any custom parsers, functions or tables, so that you validate no incorrect syntax and execution functions properly.  If your submission requires a custom parser or function, it must be submitted with the PR.

   Testing Completed:
   - Yes/No/Need Help

_Note: If updating a detection, you must update the version field._

> Before the submission has been made, please look at running the KQL and Yaml Validation Checks locally.
> https://github.com/Azure/Azure-Sentinel#run-kql-validation-locally

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes/No/Need Help
   
   _Note: Let us know if you have tried fixing the validation error and need help._

> References: 
> - [Guidance for Detection checks](https://github.com/Azure/Azure-Sentinel#pull-request-detection-template-structure-validation-check)
> - [General contribution guidance](https://github.com/Azure/Azure-Sentinel/wiki#what-can-you-contribute-and-how-can-you-create-contributions)
> - [PR validation troubleshooting](https://github.com/Azure/Azure-Sentinel#pull-request)

-----------------------------------------------------------------------------------------------------------

### CODE REVIEW

**Comment by tomzarhin5** [comment]
@microsoft-github-policy-service agree

**Comment by v-prasadboke** [comment]
Hello @tomzarhin5, Solution is missing with solution metadata and input file. 
Function app zip is missing in the data connector folder, please add it too. 

Here's a solution which you can use for reference 
https://github.com/Azure/Azure-Sentinel/tree/master/Solutions/Agari

**Comment by tomzarhin5** [comment]
How are the aka.ms link for the function app zip created? I need to include it in my azuredeploy file

**Comment by v-prasadboke** [comment]
We will update the shortlinks and thank you for committing the required changes. We will come back to you by 14 Jun, 2024.

**Comment by tomzarhin5** [comment]
Hi, I wanted to kindly remind you about this PR. You mentioned you would get back to me by June 14, 2024. I am looking forward to your response and any further instructions or updates. Thank you

**Comment by v-prasadboke** [comment]
Apologies for the inconvenience, Had too much on my plate. And was on leave yesterday. 
I'll have this on my plate and will be providing you an update by 19 Jun, 2024 EOD

File: Solutions/TransmitSecurity/Data Connectors/requirements.txt
Comment (v-prasadboke) [inline Solutions/TransmitSecurity/Data Connectors/requirements.txt:1]: python libraries mentioned in the requirements.txt are missing from zip. Please go through the zip structure of Solution mentioned earlier

**Comment by v-prasadboke** [comment]
Please add a custom table named TransmitSecurityEvents_CL at location 
https://github.com/Azure/Azure-Sentinel/tree/master/.script/tests/KqlvalidationsTests/CustomTables

Share sample data to test the content of the solution. 

Share invocation logs of Function app

File: Solutions/TransmitSecurity/SolutionMetadata.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+	"publisherId": "transmitsecurity",
+	"offerId": "transmitsecurity",
```
Comment (v-prasadboke) [inline Solutions/TransmitSecurity/SolutionMetadata.json:3]: Offer id should contain keyword sentinel. 

Please go through this readme for guidelines 

image


https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/README.md

Comment (tomzarhin5) [reply Solutions/TransmitSecurity/SolutionMetadata.json]: Resolved. Changed it to 
	"offerId": "microsoft-sentinel-solution-transmitsecurity",
Thank you

**Comment by tomzarhin5** [comment]
I have added two tables to :
https://github.com/Azure/Azure-Sentinel/tree/master/.script/tests/KqlvalidationsTests/CustomTables
name:
TransmitSecurityUserActivity_CL
TransmitSecurityAdminActivity_CL

Also I have deleted the parser as it is not giving any added value.
Where should i attach the sample data and invocation logs?

Thanks for your review

**Comment by v-prasadboke** [comment]
You can share screenshots for invocation logs here and sample data add it at this location 
https://github.com/Azure/Azure-Sentinel/tree/master/Sample%20Data/Custom

**Comment by tomzarhin5** [comment]
Hi there, all the required changes has been done. Please review again. Thank you

**Comment by v-prasadboke** [comment]
Hello @tomzarhin5, Can you provide me write access to your branch. Unable to contribute to this branch.

**Comment by tomzarhin5** [comment]
@v-prasadboke Unfortunately, this is not possible. This PR was made from my company’s forked repository, so it is managed by my company.
What is the reason for your request?

File: Sample Data/Custom/TransmitSecurity/TransmitSecurityInvocationLogExample.png
Comment (v-prasadboke) [inline Sample Data/Custom/TransmitSecurity/TransmitSecurityInvocationLogExample.png:1]: And this image as well

**Comment by v-prasadboke** [comment]
Ok @tomzarhin5 no problem. Try to repackage the solution using V3 tool. 
Please follow this Readme for using V3 tool. 

https://github.com/Azure/Azure-Sentinel/blob/master/Tools/Create-Azure-Sentinel-Solution/V3/README.md

**Comment by tomzarhin5** [comment]
Hi, @v-prasadboke. 
I have packed this solution using the v3 tool. 
You can see the version in the Package folder is 3.0.0.zip with all other components:
createUiDefinition.json
mainTemplate.json
and testParameters.json.
The solution i have created is located in the Data folder.

**Comment by v-prasadboke** [comment]
Thanks for working on it. I'll test the solution and provide you an update by 1 July, 2024

**Comment by v-prasadboke** [comment]
Hello @tomzarhin5, This branch behind master branch. Can you update this from master once.

**Comment by tomzarhin5** [comment]
Hi @v-prasadboke, There were no conflicts, so why the manual merge? Anyway, I have done it. Please let me know if anything else is required.

**Comment by v-prasadboke** [comment]
> @v-prasadboke Unfortunately, this is not possible. This PR was made from my company’s forked repository, so it is managed by my company. What is the reason for your request?

I had some commits in my local which I wanted to push. 
Can you please provide me if possible

**Comment by v-prasadboke** [comment]
> Hi @v-prasadboke, There were no conflicts, so why the manual merge? Anyway, I have done it. Please let me know if anything else is required.

I'm not able to update the branch from master too as I dont have write access to the same

**Comment by tomzarhin5** [comment]
> > @v-prasadboke Unfortunately, this is not possible. This PR was made from my company’s forked repository, so it is managed by my company. What is the reason for your request?
> 
> I had some commits in my local which I wanted to push. Can you please provide me if possible

I’m sorry, but I don’t have the necessary permissions, and my company has not approved it. However, we can merge another branch into this existing one. Could you create a new branch from this branch and let me know its name? I will merge it into this one.

**Comment by v-prasadboke** [comment]
Hello @tomzarhin5, Function app is missing with Python packages which are mentioned in requirements.txt 
Please go through function app zip of this Solution
https://github.com/Azure/Azure-Sentinel/tree/master/Solutions/Box/Data%20Connectors

Also is this Solution Microsoft supported? if not please revise the SolutionID and PublisherID once from azure marketplace

Also can you share invocation logs screenshot.

**Comment by tomzarhin5** [comment]
Hi @v-prasadboke,

I have reviewed the requirements for the “Box” solution and believe I have included all the necessary components. In my script, I am only using the python-requests library, which is mentioned in the __init__.py file. I have installed the solution, and it works fine for me. Could you specify which specific Python packages might be missing?

Regarding the Microsoft solution, I followed the Microsoft guide for preparing and creating the SolutionMetadata.json, so I hope this has been done correctly.

As for the invocation log, I have already shared it, and you instructed me to remove it.

Originally posted by @v-prasadboke in https://github.com/Azure/Azure-Sentinel/pull/10619#discussion_r1652774570

**Comment by v-prasadboke** [comment]
Hello @tomzarhin5 Can we get on a call for this. 
Please share your availability and time zone so that we can plan for the same. 
My mail id is v-prasadboke@microsoft.com 
You can ping me on teams too if needed

**Comment by v-prasadboke** [comment]
Hello @tomzarhin5 Please proceed with the changes discussed in the call.

**Comment by tomzarhin5** [comment]
Hi @v-prasadboke,

Thank you for your time today.

I have updated the zip file to include the Python packages folder. I have also checked, and we do have a publisherId (“transmitsecurity”), so all fields in the SolutionMetadata have been filled.

Thank you.

**Review by v-prasadboke — APPROVED** [review]

**Review by v-atulyadav — APPROVED** [review]

**Pull request merged**